Out of source build broken

The following used to work
```
cd vermont
mkdir build
cd build
cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_INSTALL_PREFIX=/tmp -DCMAKE_BUILD_TYPE=RelWithDebInfo -DSUPPORT_JOURNALD=ON -DSUPPORT_DTLS=OFF -DSUPPORT_ZMQ=OFF ../
make test
```

it now results in the following error:

```
Running tests...
Test project /home/user1/work/LEGACY/vermont/build
    Start 1: vermont_cppcheck_test
1/9 Test #1: vermont_cppcheck_test ............   Passed    0.22 sec
    Start 2: core_cppcheck_test
2/9 Test #2: core_cppcheck_test ...............   Passed    0.16 sec
    Start 3: common_cppcheck_test
3/9 Test #3: common_cppcheck_test .............   Passed    3.18 sec
    Start 4: ipfixlolib_cppcheck_test
4/9 Test #4: ipfixlolib_cppcheck_test .........   Passed    0.77 sec
    Start 5: modules_cppcheck_test
5/9 Test #5: modules_cppcheck_test ............   Passed    5.02 sec
    Start 6: vermonttest_build
6/9 Test #6: vermonttest_build ................   Passed    0.18 sec
    Start 7: vermont
7/9 Test #7: vermont ..........................Child aborted***Exception:   0.26 sec
    Start 8: example_2
Could not find executable example_code_2
Looked in the following places:
example_code_2
example_code_2
Release/example_code_2
Release/example_code_2
Debug/example_code_2
Debug/example_code_2
MinSizeRel/example_code_2
MinSizeRel/example_code_2
RelWithDebInfo/example_code_2
RelWithDebInfo/example_code_2
Deployment/example_code_2
Deployment/example_code_2
Development/example_code_2
Development/example_code_2
Unable to find executable: example_code_2
8/9 Test #8: example_2 ........................***Not Run   0.00 sec
    Start 9: mtutest
Could not find executable mtutest
Looked in the following places:
mtutest
mtutest
Release/mtutest
Release/mtutest
Debug/mtutest
Debug/mtutest
MinSizeRel/mtutest
MinSizeRel/mtutest
RelWithDebInfo/mtutest
RelWithDebInfo/mtutest
Deployment/mtutest
Deployment/mtutest
Development/mtutest
Development/mtutest
Unable to find executable: mtutest
9/9 Test #9: mtutest ..........................***Not Run   0.00 sec

67% tests passed, 3 tests failed out of 9

Total Test time (real) =   9.80 sec

The following tests FAILED:
	  7 - vermont (Child aborted)
	  8 - example_2 (Not Run)
	  9 - mtutest (Not Run)
Errors while running CTest
make: *** [Makefile:130: test] Error
```

which appears to be due to:
```
cd /home/user1/work/LEGACY/vermont/build/src/tests/vermonttest
./vermonttest
Vermont Testsuite, testing ...
27/05 11:24:10 CRITICAL: Could not open config dir
terminate called after throwing an instance of 'std::runtime_error'
  what():  : Could not open config dir
Aborted (core dumped)
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -225,4 +225,4 @@ matrix:
 before_install:
   - eval "${MATRIX_EVAL}"
 
-script: cmake -DCMAKE_INSTALL_PREFIX=/tmp -DCMAKE_BUILD_TYPE="$BUILD_TYPE" -DSUPPORT_MYSQL="$MYSQL" -DSUPPORT_POSTGRESQL=ON -DSUPPORT_JOURNALD=ON -DSUPPORT_DTLS="$DTLS" -DSUPPORT_ZMQ="$ZMQ" . && make -k && make test && make install
+script: mkdir build && cd build && cmake -DCMAKE_INSTALL_PREFIX=/tmp -DCMAKE_BUILD_TYPE="$BUILD_TYPE" -DSUPPORT_MYSQL="$MYSQL" -DSUPPORT_POSTGRESQL=ON -DSUPPORT_JOURNALD=ON -DSUPPORT_DTLS="$DTLS" -DSUPPORT_ZMQ="$ZMQ" .. && make -k && make test && make install

--- a/src/tests/vermonttest/CMakeLists.txt
+++ b/src/tests/vermonttest/CMakeLists.txt
@@ -62,5 +62,5 @@ IF (JOURNALD_FOUND)
 ENDIF (JOURNALD_FOUND)
 
 ADD_TEST(vermonttest_build "${CMAKE_COMMAND}" --build ${CMAKE_BINARY_DIR} --target vermonttest --config $<CONFIG>)
-ADD_TEST(vermont vermonttest)
+ADD_TEST(vermont vermonttest -c "${CMAKE_CURRENT_SOURCE_DIR}/test_configs/")
 SET_TESTS_PROPERTIES(vermont PROPERTIES DEPENDS vermonttest_build)

--- a/src/tests/vermonttest/ConfigTester.cpp
+++ b/src/tests/vermonttest/ConfigTester.cpp
@@ -7,11 +7,12 @@
 #include <map>
 #include "modules/ipfix/aggregator/Rules.hpp"
 
-ConfigTester::ConfigTester()
+ConfigTester::ConfigTester(std::string config_dir)
+	: config_dir{config_dir}
 {
-	DIR* dir = opendir("test_configs/");
+	DIR* dir = opendir(config_dir.c_str());
 	if (!dir) {
-		ERROR("Could not open config dir");
+		ERROR("Could not open config dir %s", config_dir.c_str());
 	}
 
 	struct dirent* d;
@@ -43,7 +44,7 @@ Test::TestResult ConfigTester::execTest()
 
 void ConfigTester::testConfig(const std::string& configFile)
 {
-	std::string vermontCommand = "../../../vermont -ddddd -f test_configs/" + configFile;
+	std::string vermontCommand = "../../../vermont -ddddd -f " + config_dir + "/" + configFile;
 
 	std::string generatedOutput = "gen_output/" + configFile;
 	std::string expectedOutput = "exp_output/" + configFile;

--- a/src/tests/vermonttest/ConfigTester.h
+++ b/src/tests/vermonttest/ConfigTester.h
@@ -9,13 +9,14 @@
 class ConfigTester : public Test
 {
 public:
-	ConfigTester();
+	ConfigTester(std::string config_dir);
 	~ConfigTester();
 	virtual TestResult execTest();
 
 private:
 	void test_Rules_parseProtoPattern();
 	void testConfig(const std::string& configFile);
+	std::string config_dir;
 	std::vector<std::string> configFiles;
 };
 

--- a/src/tests/vermonttest/VermontTest.cpp
+++ b/src/tests/vermonttest/VermontTest.cpp
@@ -1,3 +1,5 @@
+#include <getopt.h>
+
 #include "VermontTest.h"
 #include "AggregationPerfTest.h"
 #include "ReconfTest.h"
@@ -15,8 +17,30 @@ int main(int argc, char* argv[])
 	printf("Vermont Testsuite, testing ...\n");
 
 	bool perftest = false;
+	int opt, option_index;
+	const char *config_dir = nullptr;
 
-	if (argc>1 && strcmp(argv[1], "-perf")==0) perftest = true;
+	static const struct option long_opts[] = {
+		{ "config-dir",  required_argument, NULL, 'c' },
+		{ "perf",     required_argument, NULL, 'p' },
+		{ NULL, 0, NULL, 0}
+	};
+
+	while ((opt = getopt_long(argc, argv, "pc:", long_opts,
+		&option_index)) != EOF) {
+		switch (opt) {
+		case 'p':
+			perftest = true;
+			break;
+		case 'c':
+			config_dir = optarg;
+			break;
+		}
+	}
+
+	if (!config_dir) {
+		ERROR("--config-dir is required");
+	}
 
 	//msg_setlevel(LOG_INFO);
 	
@@ -29,7 +53,7 @@ int main(int argc, char* argv[])
 	testSuite.add(new BloomFilterTestSuite());
 	testSuite.add(new ConnectionFilterTestSuite());
 #endif
-	testSuite.add(new ConfigTester());
+	testSuite.add(new ConfigTester(config_dir));
 
 	testSuite.run();
 

